### PR TITLE
MAJOR UPDATE! eslint - disable unused rules

### PR DESCRIPTION
**Pull request by tundraray**
- Disabled some TS rules after updating a component
- Added a few new rules: 
   `@typescript-eslint/member-ordering` 
   `@typescript-eslint/no-invalid-void-type`
- Update rule prefer-destructuring

### CODE REVIEW

File: packages/eslint-plugin-xcritical/package.json [hunk 1]
```diff
@@ -46,14 +46,11 @@
 		"babel-eslint": "10.1.0",
 		"eslint": "6.8.0",
 		"eslint-config-airbnb": "18.1.0",
-		"eslint-config-prettier": "6.11.0",
```
Comment (ISp1rit) [inline packages/eslint-plugin-xcritical/package.json:49]: This ones were unused?

- eslint-config-prettier
- eslint-plugin-prettier
- prettier

Comment (tundraray) [reply packages/eslint-plugin-xcritical/package.json:49]: Yep

File: packages/eslint-plugin-xcritical/src/rules/typescript.js [hunk 2]
```diff
@@ -15,6 +15,29 @@ const typescriptRules = {
   '@typescript-eslint/no-type-alias': 'off',
   '@typescript-eslint/no-explicit-any': 'warn',
   '@typescript-eslint/quotes': ['error', 'single'],
-  '@typescript-eslint/explicit-function-return-type': ['error', { allowExpressions: true, allowTypedFunctionExpressions: true, allowHigherOrderFunctions: true }],
+  '@typescript-eslint/no-use-before-define': ['error', { functions: false }],
+  '@typescript-eslint/no-unnecessary-condition': ['error', { ignoreRhs: true }],
+  '@typescript-eslint/no-inferrable-types': 0, // this is a temporary solution due to problems in TS (see: https://github.com/microsoft/TypeScript/pull/30593)
```
Comment (ISp1rit) [inline packages/eslint-plugin-xcritical/src/rules/typescript.js:20]: Is it still necessary?

File: packages/eslint-plugin-xcritical/src/rules/typescript.js [hunk 3]
```diff
@@ -15,6 +15,29 @@ const typescriptRules = {
...
+  '@typescript-eslint/no-non-null-assertion': ['warn'],
+  '@typescript-eslint/prefer-readonly-parameter-types': 0,
+  '@typescript-eslint/no-unsafe-assignment': 0,
+  '@typescript-eslint/no-unsafe-call': 0,
+  '@typescript-eslint/no-unsafe-return': 0,
+  '@typescript-eslint/no-unsafe-member-access': 0,
+  '@typescript-eslint/explicit-module-boundary-types': ['warn'],
+  '@typescript-eslint/init-declarations': 0,
+  '@typescript-eslint/default-param-last': ['warn'],
+  '@typescript-eslint/no-non-null-asserted-optional-chain': ['warn'],
```
Comment (ISp1rit) [inline packages/eslint-plugin-xcritical/src/rules/typescript.js:41]: How about adding this: `'@typescript-eslint/no-dynamic-delete': 0`?

File: packages/eslint-plugin-xcritical/src/rules/typescript.js (hunk 2 above)
Comment (tundraray) [reply packages/eslint-plugin-xcritical/src/rules/typescript.js:20]: Sounds like it's necessary. I mean this rule should be enabled

File: packages/eslint-plugin-xcritical/src/rules/typescript.js (hunk 3 above)
Comment (tundraray) [reply packages/eslint-plugin-xcritical/src/rules/typescript.js:41]: What you mean?

**Pull request merged**